src/PostgREST/RangeQuery.hs

```diff
@@ -38,12 +41,13 @@ rangeParse range = do
       rangeIntersection lower upper
     Nothing -> rangeGeq 0
 
-rangeRequested :: RequestHeaders -> NonnegRange
-rangeRequested = rangeParse . fromMaybe "" . lookup hRange
+rangeRequested :: RequestHeaders -> Maybe NonnegRange
```

Comment:
Rather than `Maybe NonnegRange` what about using `NonnegRange` with a value of `fullRange` in place of `Nothing`?